### PR TITLE
Remove user-facing-ish asserts

### PR DESCRIPTION
**Pull request by adeak**
Ideally assertions shouldn't be used to validate input, and they shouldn't normally be tripped by end-users. I've removed what I believe were the last few assertions that should have been something else. I've also snuck in some local style fixups.

I can also see a reason for leaving all these assertions in their place, because they probably aren't strictly user-facing for various reasons. So feel free to dispute any or all of these changes.

1. An assertion in the `spherical.py` plotting example: 
    * pro: this is defined in a helper function, and it would probably not hurt to follow best practices here
    * con: this is just an example so assertions should generally be OK (I did leave the rest alone)
  
2. A bunch of assertions in `renderers.Renderers`:
    * pro: these are actual cases of input validation that are probably better to change
    * con: this module and class seems undocumented, so perhaps users aren't expected to create them in the first place

3. An assertion in `SphericalArcFromNormal`: I'm on the fence about this one.
    * pro: Ideally this should go in a test (however the `pv.wrap` call discards the original source and we're only left with a `PolyData` outside, so we can't test this).
    * con: this is not input validation, but rather the kind of "this should never happen" check that assertions should be used for
    * pro again: it doesn't test pyvista, it tests VTK. If we prefer a wild (and non-guaranteed) `AssertionError` if the impossible happens in VTK, we should add this assertion back in.

### CODE REVIEW

**Comment by adeak** [comment]
On closer inspection it seems to me that `Plotter` just passes on `groups` and `row/col_weights` to `Renderers`, so that's definitely an assert we have to change.

But this also raised a few related issues:
  1. we are missing a few test cases for invalid weight sizes (hence the decrease in coverage)
  2. `groups` and `row/col_weights` seem to be undocumented! The only mention is in https://dev.pyvista.org/examples/02-plot/multi-window.html.
  
Both need fixing. The question is whether I should add these here or in a separate PR.

File: examples/02-plot/spherical.py [hunk 1]
```diff
@@ -34,7 +35,8 @@ def _cell_bounds(points, bound_position=0.5):
     >>> cell_bounds(a)
     array([-1.25, -0.75, -0.25,  0.25,  0.75,  1.25,  1.75,  2.25])
     """
-    assert points.ndim == 1, "Only 1D points are allowed"
+    if points.ndim != 1:
+        raise ValueError("Only 1D points are allowed.")
```
Comment (akaszynski) [inline examples/02-plot/spherical.py:39]: I just found google style guide talking about how [you shouldn't use exclamation points](https://developers.google.com/style/exclamation-points).  Thanks for changing that.

**Comment by akaszynski** [comment]
I agree with the general premise that it's bad style to raise `AssertionError` when validating inputs.

> we are missing a few test cases for invalid weight sizes (hence the decrease in coverage)
I'd cover them here since PRs shouldn't decrease coverage and those lines of code won't be tested (unlike asserts).

> groups and row/col_weights seem to be undocumented! The only mention is in https://dev.pyvista.org/examples/02-plot/multi-window.html.

Separate PR.  That's outside the scope of this one and should be a relatively easy fix.

File: examples/02-plot/spherical.py (hunk 1 above)
Comment (adeak) [reply examples/02-plot/spherical.py:39]: My pleasure, but I don't think I changed any of that :smile:

**Review by MatthewFlamm — APPROVED** [review]
Looks good to me.

I have no opinion on the assert in the Example. I think it can go either way.  The assert is shorter, which is what I tend to use in my own development.  But, I can appreciate the value in the PR version.

In regards, to the assert in `CircularArcFromNormal`.  I agree that it is only testing `vtk`.  This is not a common pattern from my quick browsing in pyvista.   I think it is best to be consistent, which in this case, is to remove it.

**Comment by adeak** [comment]
Thanks for the reviews @akaszynski and @MatthewFlamm.

> The assert is shorter, which is what I tend to use in my own development.

I do that too with interactive scripts, so I mostly apply this to functions that are meant to be reused. There are several assertions in examples that I didn't touch for this reason. This was the only exception because it's in a helper function that might end up being reused by users.

**Pull request merged**